src/mysql/types.cr

```diff
@@ -154,9 +158,71 @@ abstract struct MySql::Type
     end
   end
   decl_type LongLong, 0x08u8, ::Int64
-  decl_type Int24, 0x09u8
-  decl_type Date, 0x0au8
-  decl_type Time, 0x0bu8
+  decl_type Int24, 0x09u8, ::Int32
+
+  def self.datetime_read(packet)
+    pkt = packet.read_byte!
+    return ::Time.new(0) if pkt < 1
+    year = packet.read_fixed_int(2).to_i32
+    month = packet.read_byte!.to_i32
+    day = packet.read_byte!.to_i32
+    return ::Time.new(year, month, day) if pkt < 6
+    hour = packet.read_byte!.to_i32
+    minute = packet.read_byte!.to_i32
+    second = packet.read_byte!.to_i32
+    return ::Time.new(year, month, day, hour, minute, second) if pkt < 8
+    ms = packet.read_int.to_i32 / 1000 # returns microseconds, time only supports milliseconds
+    return ::Time.new(year, month, day, hour, minute, second, ms)
+  end
+
+  def self.datetime_write(packet, v : ::Time)
+    packet.write_blob UInt8.slice(v.year.to_i16, v.year.to_i16/256, v.month.to_i8, v.day.to_i8, v.hour.to_i8, v.minute.to_i8, v.second.to_i8, v.millisecond*1000, v.millisecond*1000/256, v.millisecond*1000/65536)
+  end
+
+  decl_type Date, 0x0au8, ::Time do
+    def self.write(packet, v : ::Time)
+      self.datetime_write(packet, v)
+    end
+
+    def self.read(packet)
+      self.datetime_read(packet)
+    end
+
+    def self.parse(str : ::String)
+      raise "TextProtocol::Time not implemented"
```

Comment:
The text protocol needs to be implemented to support unprepared statement. in master currently there is an implementation for DateTime for this.